helpers/notifier.py

```diff
@@ -56,7 +57,12 @@ def build_message(self) -> str:
     async def notify(
         self,
     ) -> NotifierResult:
-        self.repo_service = get_repo_provider_service(self.commit.repository)
+        try:
+            self.repo_service = get_repo_provider_service(self.commit.repository)
+        except EntityRateLimitedException as e:
+            log.warning(
+                f"Entity {e.entity_name} rate limited trying to notify. Please try again later"
```

Comment:
What do you mean "please try again later"? There's are no users looking at this.

And the exception is caught so it won't stop the task either... What do we want to do? retry the task at a later time or actually fail it?

Reply:
Yeah as I explained in the description the warning rn is a placeholder, just an example of how this would look like if I continued with the implementation.

The approach I'd like your opinion on is:

- throwing the exception in shared [here](https://github.com/codecov/shared/pull/310/files#diff-0b2eaad1f90324bc24507241c930116cca92412ec6a38973e29ce5d82134313cR69). An alternative is to throw the exception at the `get_repo_provider_service` and `get_owner_provider_service` level, wdyt?
- catching that exception everywhere the report_service/owner_service is instantiated in worker. It would be a try/catch that captures the `EntityRateLimitedException` and likely just end the task there and log something. I can add some properties when I throw the `EntityRateLimitedException` so I can log them - entity_name, ttl, etc. Maybe there's some cases where we retry if the ttl is under some amount of time, but that can likely be a later thing. 

Just wanted to hear your opinion before doing all the try/catch + tests if that didn't feel like the approach to do